point-of-sale-app/inventory/src/main/java/com/google/abmedge/inventory/InventoryController.java

```diff
@@ -50,13 +94,92 @@ public String readiness() {
-    return "Hello Anthos BareMetal - Inventory Controller";
+  @GetMapping(value = "/items", produces = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<String> items() {
+    List<Item> inventoryItems = activeItemsType.equals(ALL_ITEMS) ?
+        activeConnector.getAll() : activeConnector.getAllByType(activeItemsType);
+    String jsonString = GSON.toJson(inventoryItems, new TypeToken<List<Item>>() {
+    }.getType());
+    return new ResponseEntity<>(jsonString, HttpStatus.OK);
+  }
+
+  @PostMapping(value = "/switch/{type}")
+  public ResponseEntity<Void> switchType(@PathVariable String type) {
+    this.activeItemsType = type;
+    LOGGER
+        .info(String.format("The active inventory type has been changed to: %s", activeItemsType));
+    return new ResponseEntity<>(HttpStatus.OK);
+  }
+
+  @PutMapping(value = "/update", consumes = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<Void> update(@RequestBody List<PurchaseItem> purchaseList) {
+    boolean updated = true;
+    for (PurchaseItem pi : purchaseList) {
+      UUID itemId = pi.getItemId();
+      Optional<Item> loadedItem = activeConnector.getById(itemId);
+      if (loadedItem.isPresent()) {
+        Item i = loadedItem.get();
+        long currentQuantity = i.getQuantity();
+        if (currentQuantity >= pi.getItemCount()) {
+          long newQuantity = currentQuantity - pi.getItemCount();
+          i.setQuantity(newQuantity);
+          updated &= activeConnector.update(i);
+          LOGGER.info(String.format("Updated item '%s - %s' with new quantity '%s'",
+              itemId, i.getName(), newQuantity));
+          continue;
+        }
+        LOGGER.error(String.format(
+            "Failed to update item '%s - %s'. "
+                + "The requested count '%s' is more than whats available '%s'",
+            itemId, i.getName(), pi.getItemCount(), currentQuantity
+        ));
+      } else {
+        LOGGER.warn(String.format("Update attempt with invalid item id: '%s'", itemId));
+        updated = false;
+      }
+    }
+    if (!updated) {
+      LOGGER.error("Failed to update one or more items in the purchase list!");
+      return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+    return new ResponseEntity<>(HttpStatus.OK);
+  }
+
+  private void initInventoryItems() {
+    String inventoryList = System.getenv(INVENTORY_ITEMS_ENV_VAR);
+    if (StringUtils.isBlank(inventoryList)) {
+      LOGGER.warn(String.format(
+          "No items found under inventory list env var '%s'", INVENTORY_ITEMS_ENV_VAR));
+      return;
+    }
+    LOGGER.debug(inventoryList);
+    Yaml yaml = new Yaml(new Constructor(Inventory.class));
+    Inventory inventory = yaml.load(inventoryList);
+    inventory.getItems().forEach(i -> {
+      i.setId(UUID.randomUUID());
+      activeConnector.insert(i);
+      LOGGER.info(String.format("Inserting new item: %s", i.toString()));
+    });
+  }
+
+  private void initConnectorType() {
```

Comment:
can you add a comment here on the purpose of the inventory connector?

Reply:
Added comment here: https://github.com/GoogleCloudPlatform/anthos-edge-usecases/pull/2/commits/5f43a1ec8a1073f2e2bce71fa6298b2be8574388